pkg/rollupsmachine/machine_test.go

```diff
@@ -0,0 +1,486 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+package rollupsmachine
+
+import (
+	"encoding/hex"
+	"fmt"
+	"log"
+	"log/slog"
+	"os"
+	"testing"
+
+	"github.com/cartesi/rollups-node/pkg/emulator"
+	"github.com/cartesi/rollups-node/test/snapshot"
+
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+func init() {
+	log.SetFlags(log.Ltime)
+	slog.SetLogLoggerLevel(slog.LevelDebug)
+}
+
+const (
+	cycles          = uint64(1_000_000_000)
+	serverVerbosity = ServerVerbosityInfo
+)
+
+func payload(s string) string {
+	return fmt.Sprintf("echo '{ \"payload\": \"0x%s\" }'", hex.EncodeToString([]byte(s)))
+}
+
+// ------------------------------------------------------------------------------------------------
+
+// TestRollupsMachine runs all the tests for the rollupsmachine package.
+func TestRollupsMachine(t *testing.T) {
+	suite.Run(t, new(RollupsMachineSuite))
+}
+
+type RollupsMachineSuite struct{ suite.Suite }
+
+func (s *RollupsMachineSuite) TestLoad()    { suite.Run(s.T(), new(LoadSuite)) }
+func (s *RollupsMachineSuite) TestFork()    { suite.Run(s.T(), new(ForkSuite)) }
+func (s *RollupsMachineSuite) TestAdvance() { suite.Run(s.T(), new(AdvanceSuite)) }
+func (s *RollupsMachineSuite) TestInspect() { suite.Run(s.T(), new(InspectSuite)) }
+func (s *RollupsMachineSuite) TestCycles()  { suite.Run(s.T(), new(CyclesSuite)) }
+
+// ------------------------------------------------------------------------------------------------
+
+// Missing:
+// - "could not create the remote machine manager"
+// - "could not read iflagsY"
+// - "could not read the yield reason"
+// - machine.Close()
```

Comment:
Any expectation to add this tests on this PR?

Reply:
This has been added to the technical debt list.

Reply:
This is now being tested with the addition of the unit test suite.